diff --git a/lib/sam3x/gcc/gcc/startup_sam3xa.c b/lib/sam3x/gcc/gcc/startup_sam3xa.c
index 337a6d3c..dcf56447 100644
--- a/lib/sam3x/gcc/gcc/startup_sam3xa.c
+++ b/lib/sam3x/gcc/gcc/startup_sam3xa.c
@@ -132,7 +132,7 @@ void CAN0_Handler   ( void ) __attribute__ ((weak, alias("Dummy_Handler")));
 void CAN1_Handler   ( void ) __attribute__ ((weak, alias("Dummy_Handler")));
 
 /* Exception Table */
-__attribute__ ((section(".vectors")))
+__attribute__ ((section(".vectors"))) __attribute__((externally_visible))
 const DeviceVectors exception_table = {
 
         /* Configure Initial Stack Pointer, using linker-generated symbols */
