--- source/gcc/startup_sam3xa.c.orig	2016-06-14 14:20:43.166209461 -0400
+++ source/gcc/startup_sam3xa.c	2016-06-14 14:00:57.497137169 -0400
@@ -129,7 +129,7 @@
 void CAN1_Handler       ( void ) __attribute__ ((weak, alias("Dummy_Handler")));
 
 /* Exception Table */
-__attribute__ ((section(".vectors")))
+__attribute__ ((section(".vectors"))) __attribute__((externally_visible))
 const DeviceVectors exception_table = {
 
 	/* Configure Initial Stack Pointer, using linker-generated symbols */
